app/assets/javascripts/account/channel_icon_select.js

```diff
@@ -0,0 +1,20 @@
+function formatState(state) {
+  if (!state.id) {
+    return state.text;
+  }
+
+  var baseUrl = "/assets/channel_icons";
+  var $state = $(
+    '<span value="' + state.text.toUpperCase() + '"}><img src="' + baseUrl + '/' + state.text.toLowerCase() + '.svg" width="30px"/> ' + state.text + '</span>'
+  );
+  return $state;
+};
+
+$(document).ready(function() {
+  $('.js-select2-channel-icon').select2({
+    placeholder: '-- Select Icon --',
+    width: '100%',
+    templateResult: formatState
+  });
+});
+
```

Comment:
![2020-01-12T22:41:08](https://user-images.githubusercontent.com/33513522/72225310-d563c300-358c-11ea-99e9-037de74ac31c.png)

Reply:
this may not work in production, with digest assets enabled

Reply:
just pass an icon url in JSON response and use it
`return state.icon ? state.icon : state.text` etc.